ReflectiveBoundTexture.go

```diff
@@ -0,0 +1,259 @@
+//
+// Parameters:
+//   - width: The width of the image. If set to -1, it will use the available content region width.
+//   - height: The height of the image. If set to -1, it will use the available content region height.
+//   - options: The options for rendering the image, including UV coordinates, tint color, and border color.
+func (i *ReflectiveBoundTexture) ImguiImageV(width, height float32, options ImguiImageVOptionStruct) {
+	size := imgui.Vec2{X: width, Y: height}
+
+	if size.X == -1 {
+		rect := imgui.ContentRegionAvail()
+		size.X = rect.X
+	}
+
+	if size.Y == -1 {
+		rect := imgui.ContentRegionAvail()
+		size.Y = rect.Y
+	}
+
+	imgui.ImageV(i.Texture().ID(), size, options.Uv0, options.Uv1, options.TintCol, options.BorderCol)
+}
+
+// ImguiImageButtonV renders the ReflectiveBoundTexture as an image button in ImGui with additional options.
+//
+// Parameters:
+//   - id: The ID of the image button.
+//   - width: The width of the image button. If set to -1, it will use the available content region width.
+//   - height: The height of the image button. If set to -1, it will use the available content region height.
+//   - options: The options for rendering the image button, including UV coordinates, tint color, and border color.
+func (i *ReflectiveBoundTexture) ImguiImageButtonV(id string, width, height float32, options ImguiImageVOptionStruct) {
+	size := imgui.Vec2{X: width, Y: height}
+
+	if size.X == -1 {
+		rect := imgui.ContentRegionAvail()
+		size.X = rect.X
+	}
+
+	if size.Y == -1 {
+		rect := imgui.ContentRegionAvail()
+		size.Y = rect.Y
+	}
+
+	imgui.ImageButtonV(id, i.Texture().ID(), size, options.Uv0, options.Uv1, options.TintCol, options.BorderCol)
+}
+
+// unbind releases the texture associated with the ReflectiveBoundTexture from the backend.
+func (i *ReflectiveBoundTexture) unbind() {
+	if i.tex != nil {
+		Context.Backend().DeleteTexture(i.tex.ID())
+		i.tex = nil
+	}
+}
+
+// bind creates a new texture from the RGBA surface and assigns it to the ReflectiveBoundTexture.
+func (i *ReflectiveBoundTexture) bind() {
+	NewTextureFromRgba(i.Surface, func(tex *Texture) {
+		i.tex = tex
+	})
+}
+
+// GetSurfaceWidth returns the width of the RGBA surface.
+func (i *ReflectiveBoundTexture) GetSurfaceWidth() int {
+	return i.Surface.Bounds().Dx()
+}
+
+// GetSurfaceHeight returns the height of the RGBA surface.
+func (i *ReflectiveBoundTexture) GetSurfaceHeight() int {
+	return i.Surface.Bounds().Dy()
+}
+
+// GetSurfaceSize returns the size of the RGBA surface as an image.Point.
+func (i *ReflectiveBoundTexture) GetSurfaceSize() image.Point {
+	return i.Surface.Bounds().Size()
+}
+
+// Texture commits any pending changes to the RGBA surface and returns the associated texture.
+func (i *ReflectiveBoundTexture) Texture() *Texture {
+	i.commit()
+	return i.tex
+}
+
+// ID commits any pending changes and returns the ImGui TextureID of the associated texture.
+func (i *ReflectiveBoundTexture) ID() imgui.TextureID {
```

Comment:
Could you please rename to TextureID or something?
Just to avoid confusion - ID is a neame of a method of `(*XXXWidget).ID(id string)` that overrides widget ID for imgui.